authentication.js

```diff
@@ -0,0 +1,29 @@
+const jwt = require("jsonwebtoken");
+require("dotenv").config();
+
+const generateToken = (payload) => {
+  return jwt.sign(payload, process.env.JWT_SECRET, {
+    expiresIn: "2days",
+  });
+};
+
+const verifyToken = (request, response, next) => {
+  const authHeader = request.headers["authorization"];
+  const token = authHeader?.split(" ")[1];
+
+  if (!token) {
+    return response.status(401).send({ error: "No token provided" });
+  } else {
+    jwt.verify(token, process.env.JWT_SECRET, (err, res) => {
+      if (err) {
+        return response.status(403).send({ error: err.message });
```

Comment:
Bad authentication should return `401`.
`403` is reserved for tokens that are unauthorized to access a particular resource (say, a file that is owned by user B cannot be accessed by user A).